cocotb/triggers.py

```diff
@@ -735,3 +735,46 @@ def _wait(self):
         for _ in range(self.num_cycles):
             yield trigger
         raise ReturnValue(self)
+
+
+@decorators.coroutine
+def with_timeout(triggers, timeout_time, timeout_unit=None):
+    """
+    Waits on triggers, throws an exception if it waits longer than the given time
+
+    Can be used with any expression that would otherwise appear to the right of a
+    `yield` (or `await` in Python 3) statement.
+
+    Usage:
+
+    .. code-block:: python
+
+        yield with_timeout(coro, 100, 'ns')
+        yield with_timeout([coro, event.wait(), ...], 100, 'ns')
+
+    Args:
+        triggers (cocotb_yield_expr):
+            Any expression that could be right of a yield statement in cocotb
+        timeout_time (int):
+            Time duration
+        timeout_unit (str, optional):
+            Units of duration, accepts any values that :class:`cocotb.triggers.Timer` does
+
+    Returns:
+        First trigger that completed if timeout did not occur.
+
+    Raises:
+        :exc:`SimTimeoutError`: If timeout occurs
+
+    .. versionadded:: 1.3
+    """
+
+    if not isinstance(triggers, list):
+        triggers = [triggers]
+
+    timeout_timer = cocotb.triggers.Timer(timeout_time, timeout_unit)
+    res = yield [timeout_timer] + triggers
```

Comment:
I know I said this before, but I think this will still work if you change this to the following:
```python
timeout_timer = cocotb.triggers.Timer(timeout_time, timeout_unit)
res = yield [timeout_timer, triggers]
```
Nested sequences are unpacked automatically, as is proven in this test:
https://github.com/cocotb/cocotb/blob/8db47936dace61ff3b9700f26c2f57a2b527ce13/tests/test_cases/test_cocotb/test_cocotb.py#L1171-L1178